New section on reflection

GeorgeMac opened the pull request:
A bit tongue in cheek, but it gets the message across.

GeorgeMac committed:
New section on reflection

A bit tongue in cheek, but it gets the message across.

GeorgeMac committed:
Update top level README with reflection header

brettbuddin commented:
Reflection for tags is probably the only amount of it I see us needing. Unless we are writing generators 🤷‍♂️

GeorgeMac commented:
Yeah @LarsFronius asked about our policy on our dependencies using reflection as well. 

Perhaps I can elaborate on the `no.` XD

GeorgeMac committed:
Update reflection package to be more description

Add thoughts on third party tools using reflection and what things actually need reflection.

GeorgeMac committed:
Update spelling to account for unmarshal
